### PR TITLE
Add ICMP duration metrics

### PR DESCRIPTION
**Pull request by SuperQ**
Add a gauge to measure various parts of the ICMP probe.

Fixes: https://github.com/prometheus/blackbox_exporter/issues/315

### CODE REVIEW

File: prober/icmp.go [hunk 1]
```diff
@@ -48,15 +48,28 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		socket      net.PacketConn
 		requestType icmp.Type
 		replyType   icmp.Type
+
+		durationGaugeVec = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "probe_icmp_duration_seconds",
+			Help: "Duration of http request by phase, summed over all redirects",
```
Comment (brian-brazil) [inline prober/icmp.go]: This needs updating

**Comment by SuperQ** [comment]
I did some testing, the Go ICMP code seems way slower than the normal C `ping` program.  For example, over my local ethernet:

```
10 packets transmitted, 10 received, 0% packet loss, time 9218ms
rtt min/avg/max/mdev = 0.312/0.354/0.396/0.031 ms
```

With the blackbox_exporter:
```
probe_icmp_duration_seconds{phase="rtt"} 0.000537164
probe_icmp_duration_seconds{phase="rtt"} 0.000538153
probe_icmp_duration_seconds{phase="rtt"} 0.000447007
probe_icmp_duration_seconds{phase="rtt"} 0.00060231
probe_icmp_duration_seconds{phase="rtt"} 0.000539882
probe_icmp_duration_seconds{phase="rtt"} 0.000630896
probe_icmp_duration_seconds{phase="rtt"} 0.000521305
probe_icmp_duration_seconds{phase="rtt"} 0.000569541
probe_icmp_duration_seconds{phase="rtt"} 0.000598364
probe_icmp_duration_seconds{phase="rtt"} 0.000574883
```
Go seems to add about 200us to the ping time. (average 556us)

This is still far more accurate than the full `probe_duration_seconds`.

```
probe_duration_seconds 0.001279944
probe_duration_seconds 0.000909941
probe_duration_seconds 0.000842068
probe_duration_seconds 0.000947099
probe_duration_seconds 0.000957239
probe_duration_seconds 0.001390443
probe_duration_seconds 0.001156965
probe_duration_seconds 0.001060642
probe_duration_seconds 0.00130084
probe_duration_seconds 0.0013451
```

File: prober/icmp.go (hunk 1 above)
Comment (SuperQ) [reply prober/icmp.go]: Whups, copy-pasta. :)

**Comment by SuperQ** [comment]
FYI, Travis is failing due to Go 1.11 `go fmt`.

**Comment by baiwfg2** [comment]
@brian-brazil @SuperQ Does `probe_icmp_duration_seconds` exist in current version ? I don't see it in prometheus metric list:

blackbox-exporter

**Comment by c0dyhi11** [comment]
> @brian-brazil @SuperQ Does `probe_icmp_duration_seconds` exist in current version ? I don't see it in prometheus metric list:
Same here...

**Comment by zhangzheng1019** [comment]
@SuperQ @brian-brazil  Similarly, I don’t have an ICMP indicator here.
image
image

**Pull request merged**